### DISCUSSION TITLE
ea_next_gen.next_gen

### DISCUSSION THREAD

**Discussion by nim-hrkn** [Tomoki-YAMASHITA/CrySPY#24] [General]
ea_next_gen.next_gen()に渡している構造は過去全ての構造に見えますが、今のstageの構造だけですか、過去全ての構造ですか。

**Comment by nim-hrkn** [comment]
全部ですね。

**Comment by Tomoki-YAMASHITA** [comment]
ctrl_job.py から ea_next_gen.next_gen() を呼び出すときは全ての構造データを渡しているのですが，
CrySPY/EA/ea_next_gen.py の最初の方

``` python
    # ---------- current generation
    c_rslt = rslt_data[rslt_data['Gen'] == gen]
    c_fitness = c_rslt['E_eV_atom'].to_dict()    # {ID: energy, ...}
```

ここで，現在の世代の結果だけ抜き出しています．
rslt_dataがpandasの形式で保存されているエネルギーなどのデータでそこから現在の世代のデータをc_rsltとして抜き出しています．
IDとエネルギーの辞書形式にc_fitnessとして直して
この中から親を選んでいます．

**Reply by nim-hrkn** [reply]
ありがとう。助かります。
